### PR TITLE
Fix #4248

### LINKED ISSUE #4248: Adding a selected option should update container text
## Steps to reproduce the issue

1. Instanciate select2 on `<select><option value="x">y</option></select>`
2. Append `<option value="bar" selected>foo</option>` to non-multiple select

(Or use add-another popup in django admin).

## Expected behavior and actual behavior

Expected: value updates to "bar", container text updates to "foo"
Actual: value updates to "bar" but container text is still "y".

## Environment

Browsers

- [ ] Google Chrome
- [ ] Mozilla Firefox
- [ ] Internet Explorer

Operating System

- [ ] Linux

Libraries

- jQuery version: vendorized from select2 master
- Select2 version: master

## Isolating the problem

A test case is provided in a PR.

### PR DESCRIPTION
**Pull request by jpic**
This pull request includes a

- [x] Bug fix

The following changes were made

- Added test case
- Observe subtree for added options and sync the container text

Issue: #4248

### CODE REVIEW

**Comment by jpic** [comment]
The new test fails as expected, still trying to fix it, but there might be a design decision to make so I'm not really sure. Should _syncAttributes be extended to support this ? should a new _sync method be added ? should we reuse the same observer and enable subtree: true on it for performance or make another observer ?

**Comment by jpic** [comment]
Ready for review B)

File: tests/integration/select2-methods.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* globals start, stop, setTimeout */
```
Comment (jpic) [inline tests/integration/select2-methods.js]: Sorry, but apparently we don't have [assert.async()](https://api.qunitjs.com/async/) in our version of QUnit !

**Comment by kevin-brown** [comment]
I've mentioned it on other channels (IRC, specifically) that this pull request has issues because it bypasses the formatting of selections. This is because instead of triggering `selection:update` with the new selection, it just sets the selection directly. This also only works on single selects, partially because that's the easiest to design for, but it results in this being inconsistent.

As a result, I'm finally going to formally reject this pull request. Tonight I'll take a shot at doing this in a way that works with the Select2 event system instead of bypassing it.

**Comment by jpic** [comment]
Perhaps you could reuse the test case though ?

**Pull request closed without merging**